lib/models/device.coffee

```diff
@@ -1345,6 +1345,41 @@ getDeviceModel = (deps, opts) ->
 						force: Boolean(options.force)
 		.asCallback(callback)
 
+	###*
+	# @summary Get the supervisor state on a device
+	# @name getSupervisorState
+	# @public
+	# @function
+	# @memberof balena.models.device
+	#
+	# @param {String|Number} uuidOrId - device uuid (string) or id (number)
+	# @returns {Promise}
+	#
+	# @example
+	# balena.models.device.getSupervisorState('7cf02a6').then(function(state) { console.log(state); });;
+	#
+	# @example
+	# balena.models.device.getSupervisorState(123).then(function(state) { console.log(state); });;
+	#
+	# @example
+	# balena.models.device.getSupervisorState('7cf02a6', function(error) { if (error) throw error; });
+	###
+	exports.getSupervisorState = (uuidOrId, callback) ->
+
+		exports.get uuidOrId,
+			$select: 'uuid'
+		.then ({ uuid }) ->
+			return request.send
+				method: 'POST'
+				url: '/supervisor/v1/device'
+				baseUrl: apiUrl
+				body:
+					uuid: uuid
+					method: 'GET'
+		.get('body')
+		.catch(notFoundResponse, treatAsMissingDevice(uuidOrId))
```

Comment:
Just to be clear, `exports.get()` will actually reject with a BalenaDeviceNotFound. The rest of the methods that use `treatAsMissingDevice` are using the plain `getId` which isn't checking the API if the parameter provided is a number, and that's why they definitely need the `treatAsMissingDevice`.
This shouldn't be necessary, but I'm fine with it, as far as the `'/supervisor/v1/device'` doesn't return a 404 in any other case than the device not being in the DB.

Reply:
You will have to remove this line, since it is causing the `it 'should be rejected if the device exists but is inaccessible'` test case to fail.